xarray/test/test_dataarray.py

```diff
@@ -1242,6 +1243,107 @@ def test_groupby_first_and_last(self):
+            da.rolling(time=7, x=2)
+        with self.assertRaisesRegexp(ValueError, 'window must be > 0'):
+            da.rolling(time=-2)
+
+    @requires_bottleneck
+    def test_rolling_wrapped_bottleneck(self):
+        import bottleneck as bn
+
+        da = self.make_rolling_example_array()
+
+        # Test all bottleneck functions
+        rolling_obj = da.rolling(time=7)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using min_periods
+        rolling_obj = da.rolling(time=7, min_periods=1)
+        for name in ('sum', 'mean', 'std', 'min', 'max'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0,
+                                              min_count=1)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using center=False
+        rolling_obj = da.rolling(time=7, center=False)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # Using center=True
+        rolling_obj = da.rolling(time=7, center=True)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # catching invalid args
+        with self.assertRaisesRegexp(ValueError, 'Rolling.median does not'):
+            da.rolling(time=7, min_periods=1).median()
+
+    def test_rolling_pandas_compat(self):
+        s = pd.Series(range(10))
+        da = DataArray.from_series(s)
+
+        for center in (False, True):
+            for window in [1, 2, 3, 4]:
+                for min_periods in [None, 0, 1, 2]:
+                    if min_periods is not None and window < min_periods:
+                        min_periods = window
+                    s_rolling = pd.rolling_mean(s, window, center=center,
+                                                min_periods=min_periods)
+                    da_rolling = da.rolling(index=window, center=center,
+                                            min_periods=min_periods).mean()
+                    # pandas does some fancy stuff in the last position,
+                    # we're not going to do that yet!
+                    np.testing.assert_allclose(s_rolling.values[:-1],
+                                               da_rolling.values[:-1])
+                    np.testing.assert_allclose(s_rolling.index,
+                                               da_rolling['index'])
+
+    def test_rolling_reduce(self):
```

Comment:
Can we make this test more comprehensive, like the one above for `test_rolling_pandas_compat`? I would be very confident in this function if we looped over every permutation of `center`, `window`, `min_periods` and aggregation function and the results were consistent.